sklearn/tests/test_common.py

```diff
@@ -87,12 +87,19 @@ def _tested_estimators():
 
 
 @parametrize_with_checks(_tested_estimators())
-def test_estimators(estimator, check):
+def test_estimators(estimator, check, request):
     # Common tests for estimator instances
     with ignore_warnings(category=(FutureWarning,
                                    ConvergenceWarning,
                                    UserWarning, FutureWarning)):
         _set_checking_parameters(estimator)
+
+        xfail_checks = _safe_tags(estimator, '_xfail_test')
+        check_name = _set_check_estimator_ids(check)
+        if xfail_checks:
+            if check_name in xfail_checks:
+                msg = xfail_checks[check_name]
+                request.applymarker(pytest.mark.xfail(reason=msg))
```

Comment:
There is probably a way to make this work for contrib projects automatically with `parametrize_with_checks`, but for now I just want something that we can use in scikit-learn as the ability to XFAIL is blocking multiple PRs.